dockers/Screenshotter/screenshotter.js

```diff
@@ -0,0 +1,241 @@
+    try {
+        // When using boot2docker, seleniumIP and katexIP are distinct.
+        seleniumIP = childProcess.execFileSync(
+            "boot2docker", ["ip"], { encoding: "utf-8" }).replace(/\n$/, "");
+        var config = childProcess.execFileSync(
+            "boot2docker", ["config"], { encoding: "utf-8" });
+        config = (/^HostIP = "(.*)"$/m).exec(config);
+        if (!config) {
+            console.error("Failed to find HostIP");
+            process.exit(2);
+        }
+        katexIP = config[1];
+    } catch(e) {
+        seleniumIP = katexIP = dockerCmd(
+            "inspect", "-f", "{{.NetworkSettings.Gateway}}", opts.container);
+    }
+    seleniumPort = dockerCmd("port", opts.container, seleniumPort);
+    seleniumPort = seleniumPort.replace(/^.*:/, "");
+    seleniumURL = "http://" + seleniumIP + ":" + seleniumPort + "/wd/hub";
+}
+if (seleniumURL) {
+    console.log("Selenium driver at " + seleniumURL);
+} else {
+    console.log("Selenium driver in local session");
+}
+
+if (!katexURL) {
+    katexURL = "http://" + katexIP + ":" + opts.katexPort + "/";
+}
+var toStrip = "http://localhost:7936/"; // remove this from testcase URLs
+
+//////////////////////////////////////////////////////////////////////
+// Wait for container to become ready
+
+var attempts = 0;
+process.nextTick(seleniumIP ? tryConnect : buildDriver);
+function tryConnect() {
+    var sock = net.connect({
+        host: seleniumIP,
+        port: +seleniumPort
+    });
+    sock.on("connect", function() {
+        sock.end();
+        attempts = 0;
+        setTimeout(buildDriver, 0);
+    }).on("error", function() {
+        if (++attempts > 50) {
+            throw new Error("Failed to connect selenium server.");
+        }
+        setTimeout(tryConnect, 200);
+    });
+}
+
+//////////////////////////////////////////////////////////////////////
+// Build the web driver
+
+var driver;
+function buildDriver() {
+    var builder = new selenium.Builder().forBrowser(opts.browser);
+    if (seleniumURL) {
+        builder.usingServer(seleniumURL);
+    }
+    driver = builder.build();
+    setSize(targetW, targetH);
+}
+
+//////////////////////////////////////////////////////////////////////
+// Set the screen size
+
+var targetW = 1024, targetH = 768;
+function setSize(reqW, reqH) {
+    return driver.manage().window().setSize(reqW, reqH).then(function() {
+        return driver.takeScreenshot();
+    }).then(function(img) {
+        img = imageDimensions(img);
+        var actualW = img.width;
+        var actualH = img.height;
+        if (actualW === targetW && actualH === targetH) {
+            process.nextTick(takeScreenshots);
+            return;
+        }
+        if (++attempts > 5) {
+            throw new Error("Failed to set window size correctly.");
+        }
+        return setSize(targetW + reqW - actualW, targetH + reqH - actualH);
+    }, check);
+}
+
+function imageDimensions(img) {
+    var buf = new Buffer(img, "base64");
```

Comment:
If you want to simplify things, we could abandon the ihdr check which is just a sanity check in any case since any valid PNG file must have `IHDR` as its first chunk. So we could just as well write

```js
var buf = new Buffer(img, "base64");
return {
    buf: buf,
    width: buf.readUInt32BE(16),
    height: buf.readUInt32BE(20)
};
```

which is even shorter than the version with the png parsing step.